sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpElementAccessTracker.cs

```diff
@@ -55,5 +55,28 @@ public CSharpElementAccessTracker(IAnalyzerConfiguration analyzerConfiguration,
                     constantValue.Value is string constant &&
                     constant == value;
             };
+
+        #region Syntax-level checking methods
+
+        public override ElementAccessCondition MatchSetter() =>
+            (context) => ((ExpressionSyntax)context.Expression).IsLeftSideOfAssignment();
```

Comment:
is this safe? is there a check before to ensure it won't be a cast error?

Reply:
I inspired this from https://github.com/SonarSource/sonar-dotnet/blob/22c93d85f30e0952788e878b8ab002faa0d958f5/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpPropertyAccessTracker.cs#L60
I assumed that trackers were already filtered on correct element type, but I actually cannot find anywhere in the code where this is done. I ll add a check beforehand unless I can find why/where there is already a check on the type

Reply:
I checked. This is safe, as tracker is registered on `ElementAccessExpression` nodes (https://github.com/SonarSource/sonar-dotnet/blob/27fdfc70a0d643ab01cd3f04288bb24a2aa11007/sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SyntaxTrackers/CSharpElementAccessTracker.cs#L36).
So all `context.Expression` elements will be a subclass of `ExpressionSyntax` in this tracker